src/v/security/audit/schemas/tests/ocsf_schemas_test.cc

```diff
@@ -1066,3 +1066,45 @@ BOOST_AUTO_TEST_CASE(make_authn_event_failure) {
     auto ser = sa::rjson_serialize(authn);
     BOOST_REQUIRE_EQUAL(ser, ::json::minify(expected));
 }
+
+BOOST_AUTO_TEST_CASE(test_ocsf_size) {
+    auto req = test_http_request();
+    req.http_headers.push_back(req.http_headers[0]); // Add another element
+
+    const ss::socket_address client_addr{ss::ipv4_addr("10.0.0.1", 12345)};
+    const ss::socket_address server_addr{ss::ipv4_addr("10.1.1.1", 23456)};
+    auto http_req = ss::http::request::make(
+      req.http_method, req.url.hostname, req.url.url_string);
+    http_req.parse_query_param();
+    http_req._client_address = client_addr;
+    http_req._server_address = server_addr;
+    http_req._version = req.version;
+    http_req._headers["User-Agent"] = req.user_agent;
+    http_req._headers["Authorization"] = "You shouldn't see this at all";
+    http_req.protocol_name = req.url.scheme;
+
+    const ss::sstring username = "test";
+    const ss::sstring service_name = "test-service";
+    const auto authn = sa::make_authentication_failure_event(
+      http_req, security::credential_user{username}, service_name, "FAILURE");
+
+    size_t estimated_size = sizeof(sa::ocsf_base_event<sa::authentication>);
+    const auto& [activity_id, auth_protocol, auth_protocol_id, dest_endpoint_host, is_cleartext, is_mfa, service, src_endpoint_host, status_id, status_detail, user]
+      = authn.equality_fields();
+    estimated_size += sizeof(activity_id);
+    estimated_size += auth_protocol.size();
+    estimated_size += sizeof(auth_protocol_id);
+    estimated_size += dest_endpoint_host.size();
+    estimated_size += sizeof(is_cleartext);
+    estimated_size += sizeof(is_mfa);
+    estimated_size += service.name.size();
+    estimated_size += src_endpoint_host.size();
+    estimated_size += sizeof(status_id);
+    estimated_size += 1
+                      + (status_detail.has_value() ? status_detail->size() : 0);
+    estimated_size += user.domain.size() + user.name.size() + user.uid.size()
+                      + sizeof(user.type_id);
+
+    BOOST_CHECK_EQUAL(estimated_size, 225);
+    BOOST_CHECK_EQUAL(authn.estimated_size(), 225);
```

Comment:
can you add in a check here to see how close the estimated size is compared to the serialized JSON object?  Is a good tolerance +/- 10%? +/- 500B?

Reply:
Check comment above